dds/DCPS/DataReaderImpl_T.h

```diff
@@ -993,6 +993,15 @@ namespace OpenDDS {
       if (!encap.to_encoding(encoding, MarshalTraitsType::extensibility())) {
         return;
       }
+
+      if (decoding_modes_.find(encoding.kind()) == decoding_modes_.end()) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR ")
+          ACE_TEXT("%CDataReaderImpl::dds_demarshal: ")
+          ACE_TEXT("Encoding kind of the received sample does not ")
+          ACE_TEXT("match the ones specified by DataReader.\n"),
+          TraitsType::type_name()));
+        return;
+      }
```

Comment:
Can you put in a debug message saying what encoding is being used for the deserialization?

Reply:
I added a debug message for this.